Modules/Quotas/test/Quotas.Application.Tests/Tests/Quotas/CreateQuotaForIdentity/HandlerTests.cs

```diff
@@ -49,6 +52,37 @@ public async Task Creates_quota_for_identity()
         ).MustHaveHappened();
     }
 
+    [Fact]
+    public async Task Create_quota_with_invalid_metric_key_throws_domain_exception()
+    {
+        // Arrange
+        var command = new CreateQuotaForIdentityCommand(IdentityAddress.Parse("id1KJnD8ipfckRQ1ivAhNVLtypmcVM5vPX4j"), "An-Invalid-Metric-Key", 5, QuotaPeriod.Month);
+        var identitiesRepository = A.Fake<IIdentitiesRepository>();
+        var metricsRepository = new FindMetricsStubRepository(new Metric(MetricKey.NumberOfSentMessages, "Number Of Sent Messages"));
+        var handler = CreateHandler(identitiesRepository, metricsRepository);
+
+        // Act
+        var exception = await Assert.ThrowsAsync<DomainException>(async () => await handler.Handle(command, CancellationToken.None));
+        exception.Message.Should().Contain("The given metric key is not supported.");
+        exception.Code.Should().Be("error.platform.quotas.unsupportedMetricKey");
+    }
+
+    [Fact]
+    public async Task Create_quota_for_non_existent_identity_throws_not_found_exception()
+    {
+        // Arrange
+        var command = new CreateQuotaForIdentityCommand(IdentityAddress.Parse("id1KJnD8ipfckRQ1ivAhNVLtypmcVM5vPX4j"), "An-Invalid-Metric-Key", 5, QuotaPeriod.Month);
+        var identitiesRepository = A.Fake<IIdentitiesRepository>();
+        A.CallTo(() => identitiesRepository.Find(A<string>._, A<CancellationToken>._, A<bool>._)).Returns((Identity)null);
+        var metricsRepository = new FindMetricsStubRepository(new Metric(MetricKey.NumberOfSentMessages, "Number Of Sent Messages"));
+        var handler = CreateHandler(identitiesRepository, metricsRepository);
+
+        // Act
+        var exception = await Assert.ThrowsAsync<NotFoundException>(async () => await handler.Handle(command, CancellationToken.None));
+        exception.Message.Should().Be("Identity not found. Make sure the ID exists and the record is not expired.");
```

Comment:
Don't check for the exact error message. This will cause lots of false positives once we change the error message even the slightest bit.
In this case, it's sufficient to check if the message starts with "Identity"